include/envoy/upstream/cluster_manager.h

```diff
@@ -339,6 +393,21 @@ class CdsApi {
 
 using CdsApiPtr = std::unique_ptr<CdsApi>;
 
+/**
+ * Abstract interface for a On-Demand CDS API provider.
+ */
+class OdCdsApi {
+public:
+  virtual ~OdCdsApi() = default;
+
+  /**
+   * File an on-demand request for a cluster.
+   */
+  virtual void updateOnDemand(const std::string& cluster_name) PURE;
```

Comment:
Porting my comment over from the other PR: I'm not sure why this is needed at this interface level. From my perspective everything should be encapsulated by the cluster manager. (See above comment)